dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -22,6 +22,36 @@ def clever
     end
   end
 
+  # POST /users/auth/maker_google_oauth2
+  def maker_google_oauth2
```

Comment:
there's a lot of heavy-lifting logic in this controller method -- can we write unit tests here? (this can totally be in a separate PR)